osbs/build/build_requestv2.py

```diff
@@ -39,19 +39,25 @@ class BaseBuildRequest(object):
       * implement `render` method which returns builds input
       * initialize proper user_params class
     """
-    def __init__(self, build_json_store, outer_template):
-        self.build_json_store = build_json_store
+    def __init__(self, osbs_api, outer_template, user_params, build_json_store=None):
         self._openshift_required_version = parse_version('3.6.0')
         self._outer_template_path = outer_template
         self._resource_limits = None
         self._template = None
 
-        self.isolated = None
         self.organization = None
-        self.osbs_api = None
-        self.scratch = None
+        self.osbs_api = osbs_api
         self.source_registry = None
-        self.user_params = None  # must be initialized in subclasses
+        self.user_params = user_params
+        if user_params and user_params.build_json_dir.value:
+            self._build_json_store = user_params.build_json_dir.value
+        else:
+            self._build_json_store = build_json_store
```

Comment:
Does `BuildRequest` need to have its own `build_json_store` attribute? Would the `user_params` one be enough?

Reply:
It's mostly a way to keep the API the same.  The build_json_store that is used is the one in user_params, but this lets you create a BuildRequest with a build_json_store, create a default user_params, and then the set_params call gives the user_params the correct build_json_store.

Reply:
Ok, that makes sense :+1: